samples/boards/nxp/adsp/rtxxx/amp_talk/remote_hifi4/boards/mimxrt700_evk_mimxrt798s_hifi4.overlay

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2025 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/dt-bindings/memory-attr/memory-attr-arm.h>
+
+/ {
+	chosen {
+		zephyr,ipc_shm_hifi4 = &sram_shared1;
+		zephyr,ipc = &mailbox_b;
+	};
+
+	mailbox_b: ipm-mbox {
+		compatible = "zephyr,mbox-ipm";
+		mboxes = <&mbox4_b 0>, <&mbox4_b 1>;
+		mbox-names = "tx", "rx";
+	};
+
+    /* Define memory regions */
+	sram_shared_mem_hifi4: memory@20584000 {
```

Comment:
In commit 1 you already added 
```
sram_shared: memory@20580000 {
    ...
    sram_shared0: memory@20580000 { ... };
    sram_shared1: memory@20584000 { ... };
};
```
Now you have 2 distinct DTS nodes describing the same physical region.
Why can the SoC nodes be reused?

Instead of re-declaring them, just extend them:
```
&sram_shared1 {
    compatible = "zephyr,memory-region", "mmio-sram";
    zephyr,memory-region = "SRAM_SHARED_HIFI4";
    zephyr,memory-attr = <(DT_MEM_ARM(ATTR_MPU_IO))>;
};
```